nextjs-metrics-viewer/app/api/analytics/services/read.ts

```
@@ -0,0 +1,58 @@
+class GetAnalyticsService {
+  constructor(private readonly __db: DatabaseConnectionHandler) {}
+
+  async perform(params: GetAnalyticsParams) {
+    try {
+      const { category, startDate, endDate } = params;
+
+      const startTimestamp = startDate ? new Date(startDate).getTime() : null;
+      const endTimestamp = endDate ? new Date(endDate).getTime() : null;
+
+      if (
+        (startDate && startTimestamp !== null && isNaN(startTimestamp)) ||
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)